protobufs/livekit_metrics.proto

```diff
@@ -0,0 +1,62 @@
+syntax = "proto3";
+
+package livekit;
+option go_package = "github.com/livekit/protocol/livekit";
+option csharp_namespace = "LiveKit.Proto";
+option ruby_package = "LiveKit::Proto";
+
+
+/*
+  Protocol used to record metrics for a specific session.
+
+  Clients send their timestamp in their own monotonically increasing time (e.g `performance.now` on JS).
+  These timestamps are then converted by the SFU to align with the session's timeline, relative to the session start time.
+  (at receive time)
+
+  A metric can be linked to a specific track by setting `track_sid`.
+*/
+
+
+// index starting at 4096 are for predefined labels (`MetricLabel`)
+enum MetricLabel {
+  AGENTS_LLM_TTFT = 0; // time to first token from LLM
+  AGENTS_STT_TTFT = 1; // time to final transcription
+  AGENTS_TTS_TTFB = 2; // time to first byte
+}
+
+message MetricsBatch {
+  // To avoid repeating string values, we store them in a separate list and reference them by index
+  // This is useful for storing participant identities, track names, etc.
+  // There is also a predefined list of labels that can be used to reference common metrics starting at 4096.
+  // E.g: for the llm TTFT, the int value would be 4096 + 0 (MetricLabel.AGENTS_LLM_TTFT)
+  // Those predefined labels are not stored in the string list
+  repeated string str_data = 1;
```

Comment:
the low values should be reserved for predefined values since we aren't likely to define more than maybe 100 of them. in large aggregate batches there could be thousands of track/participant ids.

Reply:
Agreed. Let us reserve smaller numbers for pre-defined ones.